README.md
<div align="center">
<img src="/assets/img/img-config/logo-88x88.png">
<br><br>

[![Blog-Title-Badge](https://img.shields.io/badge/-::_BIN___::-70CAC3?style=for-the-badge)](https://habijung.github.io)
<br>
</div>


# 정하빈 (Habin Jung)

<!---
<img src="assets/images/img-settings/profile-rounded-500x500.png" width="200" height="200">
--->

[![Blog-Badge](https://img.shields.io/badge/-Blog-70CAC3?logo=jekyll&style=flat)](https://habijung.github.io)&ensp;
[![Gmail-Badge](https://img.shields.io/badge/-Gmail-EA4335?logo=Gmail&logoColor=white&style=flat)](mailto:habijung0@gmail.com)&ensp;
[![GitHub-Badge](https://img.shields.io/badge/-GitHub-181717?logo=github&style=flat)](https://github.com/habijung)

**Academic Student** @[**UNIST**](https://unist.ac.kr)

Department of **Computer Science** and **Engineering**


# Blog Menu

| Menu       | Description                     |
| :--:       | ------------------------------- |
| Home       | Blog home                       |
| About      | Introduce myself & blog         |
| Posts      | All posts                       |
| Categories | Posts categories                |
| Tags       | Posts tags                      |


# Blog Update History

[:link: Blog Update History](https://habijung.github.io/history/)


# References

- [Minimal Mistakes](https://mmistakes.github.io/minimal-mistakes/)
- [Markdown Cheat Sheet](https://www.markdownguide.org/cheat-sheet/)
- [WebFX / Emoji Cheat Sheet](https://www.webfx.com/tools/emoji-cheat-sheet/)
- [Emoji Unicode List](https://unicode.org/emoji/charts/full-emoji-list.html)

---
**Updated :** 2021-08-16 21:57


_pages/history.md
---
title: "History"
permalink: /history/
layout: single-custom
author_profile: true
sidebar_category: true
---

## Blog Update History

| Name                                                                                                   | Release         |
| :---                                                                                                   | :-----:         |
| Create repository                                                                                      | 07.19.2021      |
| Apply Jekyll with [Minimal Mistakes](https://mmistakes.github.io/minimal-mistakes/){: target="_blank"} | 07.20.2021      |
| Create home page                                                                                       | 07.25.2021      |
| Update blog logo, profile                                                                              | 08.03.2021      |
| First post                                                                                             | 08.08.2021      |
| Update favicon                                                                                         | 08.12.2021      |
| Update home page URL                                                                                   | 08.13.2021      |
| Register domain [habijung.xyz](https://habijung.xyz){: target="_blank"}                                | 08.19.2021      |
| Add sidebar by categories                                                                              | 08.20.2021      |
| Register Google search console                                                                         | 08.22.2021      |
| Apply comments by [utterances](https://utteranc.es/){: target="_blank"}                                | 08.24.2021      |
| Change blog skin color #70CAC3                                                                         | 08.31.2021      |
| Apply pagination & year pages to posts page                                                            | 09.09.2021      |
| Apply blog dark mode toggle button                                                                     | 10.31.2021      |


## Blog Updating Task

2. Post CSS update log & code
3. Add statistic analytics with Google
4. Set blog theme change button


git-push.sh
#!/usr/bin/bash
# Use only remote : origin

COMMIT_FLAG=0
REMOTE_REPO=$(git remote)
WORKING_BRANCH=$(git branch --show-current)

echo -e "\n-> git pull ..."
echo "-> remote : ${REMOTE_REPO}"
echo "-> branch : ${WORKING_BRANCH}"

git pull ${REMOTE_REPO} ${WORKING_BRANCH}

echo "-> git pull complete"
echo "-> git add ..."

git add --all

echo "-> git add complete"
echo "-> git status ..."

git status

echo "-> git commit ..."

if [ $# -eq 0 ]; then
	read -p "-> Enter commit message : " COMMIT_MESSAGE
else
	COMMIT_MESSAGE=${1}
fi

echo "-> Your commit message : ${COMMIT_MESSAGE}"
read -p "-> [ Yes = 1 | No = other ] : " COMMIT_FLAG

if [ ${COMMIT_FLAG} -eq 1 ]; then

	git commit -m "${COMMIT_MESSAGE}"

	echo "-> git commit complete"
	echo "-> git push ..."

	git push ${REMOTE_REPO} HEAD:${WORKING_BRANCH}

	if [ $? = 0 ]; then
		echo -e "-> git push complete\n"
	else
		git reset --soft HEAD^
		echo "-> git push faild. Please check git error message."
	fi

else
	echo "-> git push cancel. Please try again."
fi


_plugins/post.rb
module Jekyll

    class Post
  
      # override post method in order to return categories names as slug
      # instead of strings
      #
      # An url for a post with category "category with space" will be in
      # slugified form : /category-with-space
      # instead of url encoded form : /category%20with%20space
      #
      # @see utils.slugify
      def url_placeholders
        {
            :year        => date.strftime("%Y"),
            :month       => date.strftime("%m"),
            :day         => date.strftime("%d"),
            :title       => slug,
            :i_day       => date.strftime("%-d"),
            :i_month     => date.strftime("%-m"),
            :categories  => (categories || []).map { |c| Utils.slugify(c) }.join('/'),
            :short_month => date.strftime("%b"),
            :short_year  => date.strftime("%y"),
            :y_day       => date.strftime("%j"),
            :output_ext  => output_ext
        }
      end
  
    end
  
  end
  

assets/js/js-investing/2021-08-18-dividend-portfolio.js
/* data */
const holdingsType = ['Dividends', 'Others'];
const holdingsValue = [18657.25, 42876.94];
const holdingsPercent = [30.34, 69.66];
const tickers = ['VOO','NVDA','MSFT','AAPL','TSM','KO','O','MCD','WMT','FTS','AWK','PG','TGT','COST','SRVR','GILD','SBUX','QQQ','JNJ','INTC','CSCO','NKE','PFE'];
const tickersValue = [3708.00,2180.23,2123.47,2009.15,1261.04,844.47,797.65,662.17,583.12,456.80,429.34,413.39,393.76,369.44,366.96,346.79,337.22,304.11,300.33,285.43,223.25,136.96,124.17];

/* chart : holdingsRatio */
/* setup */
const data1 = {
    labels: holdingsType,
    datasets: [{
        axis: 'y',
        fill: false,
        data: holdingsPercent,
        backgroundColor: [
            'rgba(255, 99, 132, 0.2)', // red
            'rgba(54, 162, 235, 0.2)', // blue
        ],
        borderColor: [
            'rgb(255, 99, 132)', // red
            'rgb(54, 162, 235)', // blue
        ],
        borderWidth: 1
    }]
};

/* config */
const config1 = {
    type: 'bar',
    data: data1,
    options: {
        indexAxis: 'y',
        scales: {           
            xAxes: {
                title: {
                    display: true,
                    text: 'Percent (%)',
                    align: 'end',
                    font: {
                        size: 14,
                        weight: 'bold'
                    }
                },
                position: 'top',
                suggestedMax: 100,
                ticks: {
                    beginAtZero: true,
                    stepSize: 20
                }
            }
        },
        plugins: {
            title: {
                display: true,
                text: ':: Holdings Ratio ::',
                font: {
                    size: 24,
                    weight: 'bold'
                }
            },
            legend: {
                display: false
            },
            tooltip: {
                enabled: true,
                displayColors: false,
                titleAlign: 'center',
                bodyAlign: 'right',
                callbacks: {
                    label: function(context) {
                        var value = holdingsValue[context.dataIndex];
                        var result = '$ ' + numberWithCommas(value.toFixed(2));
                        return result;
                    },
                    afterLabel: function(context) {
                        var value = context.dataset.data[context.dataIndex];
                        var percent = value.toFixed(2) + ' %';
                        return percent;
                    }
                }
            }
        }
    }
};

/* create chart */
var myChart1 = new Chart(
    document.getElementById('holdingsRatio'),
    config1
);


/* chart : dividendsJuly2021 */
/* setup */
const data2 = {
    labels: tickers,
    datasets: [{
        axis: 'y',
        fill: false,
        data: tickersValue,
        backgroundColor: [
            'rgba(255, 99, 132, 0.2)', // red
            'rgba(255, 159, 64, 0.2)', // orange
            'rgba(255, 205, 86, 0.2)', // yellow
            'rgba(75, 192, 192, 0.2)', // green
            'rgba(54, 162, 235, 0.2)', // blue
            'rgba(153, 102, 255, 0.2)', // violet
            'rgba(201, 203, 207, 0.2)' // gray
        ],
        borderColor: [
            'rgb(255, 99, 132)', // red
            'rgb(255, 159, 64)', // orange
            'rgb(255, 205, 86)', // yellow
            'rgb(75, 192, 192)', // green
            'rgb(54, 162, 235)', // blue
            'rgb(153, 102, 255)', // violet
            'rgb(201, 203, 207)' // gray
        ],
        borderWidth: 1
    }]
};

/* config */
const config2 = {
    type: 'bar',
    data: data2,
    options: {
        indexAxis: 'y',
        /* options > scales */
        scales: {
            xAxes: {
                title: {
                    display: true,
                    text: 'USD ($)',
                    align: 'end',
                    font: {
                        size: 14,
                        weight: 'bold'
                    }
                },
                position: 'top',
                ticks: {
                    beginAtZero: true,
                    stepSize: 500
                }
            }
        },
        /* options > plugins */
        plugins: {
            /* options > plugins > title */
            title: {
                display: true,
                text: ':: Dividends July 2021 ::',
                font: {
                    size: 24,
                    weight: 'bold'
                }
            },
            /* options > plugins > legend */
            legend: {
                display: false,
            },
            /* options > plugins > tooltip */
            tooltip: {
                enabled: true,
                displayColors: false,
                titleAlign: 'center',
                bodyAlign: 'right',
                /* options > plugins > tooltip > callbacks */
                callbacks: {
                    label: function(context) {
                        var value = context.dataset.data[context.dataIndex];
                        var result = '$ ' + numberWithCommas(value.toFixed(2));
                        return result;
                    },
                    afterLabel: function(context) {
                        const reducer = (accumulator, curr) => accumulator + curr;
                        var value = context.dataset.data[context.dataIndex];
                        var sum = holdingsValue.reduce(reducer);
                        var percent = (value/sum*100).toFixed(2)+' %';
                        return percent;
                    }
                } /* callbacks close */
            } /* tooltip close */
        } /* plugins close */
    } /* options close */
};

/* create chart */
var myChart2 = new Chart(
    document.getElementById('dividendsJuly2021'),
    config2
);


/* A number with commas as thousands separators
   https://stackoverflow.com/questions/2901102/how-to-print-a-number-with-commas-as-thousands-separators-in-javascript */
function numberWithCommas(x) {
    return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
}


assets/js/js-investing-log/2021-09-09-aug-2021-portfolio.js
/* data */
const holdingsType = ['Dividends', 'ETFs', 'Others'];
const holdingsValue = [14278.18, 5217.18, 42836.17];
const holdingsPercent = [22.91, 8.37, 68.72];
const tickers = ["NVDA","MSFT","AAPL","TSM","KO","O","MCD","Others"];
const tickersValue = [2180.23,2123.47,2009.15,1261.04,844.47,797.65,662.17,4400.00];


/* chart : holdingRatio */
/* setup */
const data1 = {
    labels: holdingsType,
    datasets: [{
        axis: 'y',
        fill: false,
        data: holdingsPercent,
        backgroundColor: [
            'rgba(255, 99, 132, 0.2)', // red
            'rgba(75, 192, 192, 0.2)', // green
            'rgba(54, 162, 235, 0.2)', // blue
        ],
        borderColor: [
            'rgb(255, 99, 132)', // red
            'rgba(75, 192, 192)', // green
            'rgb(54, 162, 235)', // blue
        ],
        borderWidth: 1
    }]
};

/* config */
const config1 = {
    type: 'bar',
    data: data1,
    options: {
        indexAxis: 'y',
        scales: {           
            xAxes: {
                title: {
                    display: true,
                    text: 'Percent (%)',
                    align: 'end',
                    font: {
                        size: 14,
                        weight: 'bold'
                    }
                },
                position: 'top',
                suggestedMax: 100,
                ticks: {
                    beginAtZero: true,
                    stepSize: 20
                }
            }
        },
        plugins: {
            title: {
                display: true,
                text: ':: Holdings Aug 2021 ::',
                font: {
                    size: 24,
                    weight: 'bold'
                }
            },
            legend: {
                display: false
            },
            tooltip: {
                enabled: true,
                displayColors: false,
                titleAlign: 'center',
                bodyAlign: 'right',
                callbacks: {
                    label: function(context) {
                        var value = holdingsValue[context.dataIndex];
                        var result = '$ ' + numberWithCommas(value.toFixed(2));
                        return result;
                    },
                    afterLabel: function(context) {
                        var value = context.dataset.data[context.dataIndex];
                        var percent = value.toFixed(2) + ' %';
                        return percent;
                    }
                }
            }
        }
    }
};

/* create chart */
var myChart1 = new Chart(
    document.getElementById('holdingsRatio'),
    config1
);


/* chart : dividendPortfolio */
/* setup */
const data2 = {
    labels: tickers,
    datasets: [{
        axis: 'y',
        fill: false,
        data: tickersValue,
        backgroundColor: [
            'rgba(255, 99, 132, 0.2)', // red
            'rgba(255, 159, 64, 0.2)', // orange
            'rgba(255, 205, 86, 0.2)', // yellow
            'rgba(75, 192, 192, 0.2)', // green
            'rgba(54, 162, 235, 0.2)', // blue
            'rgba(153, 102, 255, 0.2)', // violet
            'rgba(201, 203, 207, 0.2)' // gray
        ],
        borderColor: [
            'rgb(255, 99, 132)', // red
            'rgb(255, 159, 64)', // orange
            'rgb(255, 205, 86)', // yellow
            'rgb(75, 192, 192)', // green
            'rgb(54, 162, 235)', // blue
            'rgb(153, 102, 255)', // violet
            'rgb(201, 203, 207)' // gray
        ],
        borderWidth: 1
    }]
};

/* config */
const config2 = {
    type: 'bar',
    data: data2,
    options: {
        indexAxis: 'y',
        /* options > scales */
        scales: {
            xAxes: {
                title: {
                    display: true,
                    text: 'USD ($)',
                    align: 'end',
                    font: {
                        size: 14,
                        weight: 'bold'
                    }
                },
                position: 'top',
                ticks: {
                    beginAtZero: true,
                    stepSize: 500
                }
            }
        },
        /* options > plugins */
        plugins: {
            /* options > plugins > title */
            title: {
                display: true,
                text: ':: Dividends Aug 2021 ::',
                font: {
                    size: 24,
                    weight: 'bold'
                }
            },
            /* options > plugins > legend */
            legend: {
                display: false,
            },
            /* options > plugins > tooltip */
            tooltip: {
                enabled: true,
                displayColors: false,
                titleAlign: 'center',
                bodyAlign: 'right',
                /* options > plugins > tooltip > callbacks */
                callbacks: {
                    label: function(context) {
                        var value = context.dataset.data[context.dataIndex];
                        var result = '$ ' + numberWithCommas(value.toFixed(2));
                        return result;
                    },
                    afterLabel: function(context) {
                        const reducer = (accumulator, curr) => accumulator + curr;
                        var value = context.dataset.data[context.dataIndex];
                        var sum = holdingsValue.reduce(reducer);
                        var percent = (value/sum*100).toFixed(2)+' %';
                        return percent;
                    }
                } /* callbacks close */
            } /* tooltip close */
        } /* plugins close */
    } /* options close */
};

/* create chart */
var myChart2 = new Chart(
    document.getElementById('dividendPortfolio'),
    config2
);


/* A number with commas as thousands separators
   https://stackoverflow.com/questions/2901102/how-to-print-a-number-with-commas-as-thousands-separators-in-javascript */
function numberWithCommas(x) {
    return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ",");
}


assets/js/js-posts/js-daily-return/2021-10-05-oct-2021-return.js
/* data */
const daysUS = [                                                        '10/01(Fri)',
                '10/04(Mon)', '10/05(Tue)', '10/06(Wed)', '10/07(Thu)', '10/08(Fri)',
                '10/11(Mon)', '10/12(Tue)', '10/13(Wed)', '10/14(Thu)', '10/15(Fri)',
                '10/18(Mon)', '10/19(Tue)', '10/20(Wed)', '10/21(Thu)', '10/22(Fri)',
                '10/25(Mon)', '10/26(Tue)', '10/27(Wed)', '10/28(Thu)', '10/29(Fri)'];
const retValueUS = [                           0.48,
                    -1.44, 0.92,  0.76, 1.28, -0.52,
                    -0.28, 0.46,  0.75, 1.32,  1.69,
                     1.66, 0.06, -0.25, 1.41, -0.56,
                     4.17, 0.55,  1.10, 2.03,  1.01];

const daysKR = [                                                        '10/01(Fri)',
                              '10/05(Tue)', '10/06(Wed)', '10/07(Thu)', '10/08(Fri)',
                              '10/12(Tue)', '10/13(Wed)', '10/14(Thu)', '10/15(Fri)',
                '10/18(Mon)', '10/19(Tue)', '10/20(Wed)', '10/21(Thu)', '10/22(Fri)',
                '10/25(Mon)', '10/26(Tue)', '10/27(Wed)', '10/28(Thu)', '10/29(Fri)'];
const retValueKR = [                           -1.36,
                           -3.53, 0.35,  4.01, -0.73,
                           -2.76, 1.89,  2.96,  0.57,
                     0.07,  3.08, 0.02,  0.13,  0.05,
                    -0.95,  1.61, 0.31, -0.91, -0.80];



/*******************************
 *     chart : returnOctUS     *
 *******************************/
/* setup */
const dataUS = {
    labels: daysUS,
    datasets: [{
        data: retValueUS,
        fill: {
            target: 'origin',
            above: 'rgba(255, 99, 132, 0.7)',
            below: 'rgba(54, 162, 235, 0.7)'
        },
        borderColor: 'rgb(75, 192, 192)',
        hoverBackgroundColor: "rgba(0, 0, 0, 0.5)",
        tension: 0.3
    }]
};


/* config */
const configUS = {
    type: 'line',
    data: dataUS,
    options: {
        scales: {
            yAxes: {
                suggestedMax: 5,
                suggestedMin: -2,
                ticks: {
                    beginAtZero: true,
                    stepSize: 1
                }
            }
        },

        plugins: {
            title: {
                display: true,
                text: ':: [US] Daily Return Oct. 2021 ::',
                font: {
                    size: 24,
                    weight: 'bold'
                }
            },
            legend: {
                display: false
            },
            tooltip: {
                enabled: true,
                displayColors: false,
                titleAlign: 'center',
                bodyAlign: 'right',
                callbacks: {
                    label: function(context) {
                        var value = retValueUS[context.dataIndex];

                        if (value > 0)
                            return '▲ ' + value.toFixed(2) + ' %';

                        else if (value < 0)
                            return '▼ ' + Math.abs(value).toFixed(2) + ' %';

                        else
                            return '0.00 %';
                    }
                }
            }
        }
    }
};


/*******************************
 *     chart : returnOctKR     *
 *******************************/
/* setup */
const dataKR = {
    labels: daysKR,
    datasets: [{
        data: retValueKR,
        fill: {
            target: 'origin',
            above: 'rgba(255, 99, 132, 0.7)',
            below: 'rgba(54, 162, 235, 0.7)'
        },
        borderColor: 'rgb(75, 192, 192)',
        hoverBackgroundColor: "rgba(0, 0, 0, 0.5)",
        tension: 0.3
    }]
};


/* config */
const configKR = {
    type: 'line',
    data: dataKR,
    options: {
        scales: {
            yAxes: {
                suggestedMax: 5,
                suggestedMin: -4,
                ticks: {
                    beginAtZero: true,
                    stepSize: 1
                }
            }
        },

        plugins: {
            title: {
                display: true,
                text: ':: [KR] Daily Return Oct. 2021 ::',
                font: {
                    size: 24,
                    weight: 'bold'
                }
            },
            legend: {
                display: false
            },
            tooltip: {
                enabled: true,
                displayColors: false,
                titleAlign: 'center',
                bodyAlign: 'right',
                callbacks: {
                    label: function(context) {
                        var value = retValueKR[context.dataIndex];

                        if (value > 0)
                            return '▲ ' + value.toFixed(2) + ' %';

                        else if (value < 0)
                            return '▼ ' + Math.abs(value).toFixed(2) + ' %';

                        else
                            return '0.00 %';
                    }
                }
            }
        }
    }
};


/* create chart */
var myChartUS = new Chart(
    document.getElementById('returnOctUS'),
    configUS
);

var myChartKR = new Chart(
    document.getElementById('returnOctKR'),
    configKR
);


assets/js/js-daily-return/2021-09-30-sep-2021-return.js
/* data */
const days = ['09/01(Wed)', '09/02(Thu)', '09/03(Fri)', '09/06(Mon)', '09/07(Tue)', '09/08(Wed)', '09/09(Thu)', '09/10(Fri)', '09/13(Mon)', '09/14(Tue)', '09/15(Wed)', '09/16(Thu)', '09/17(Fri)', '09/20(Mon)', '09/21(Tue)', '09/22(Wed)', '09/23(Thu)', '09/24(Fri)', '09/27(Mon)', '09/28(Tue)', '09/29(Wed)', '09/30(Thu)'];
const retValue = [0.19, -0.07, 0.36, 0.0, 1.04, -0.09, -0.32, -1.23, 0.09, 0.02, 0.97, 0.09, -0.68, -2.78, 0.51, 1.27, 0.88, 0.69, -0.05, -2.63, -0.1, -0.55];


/* chart : dailyReturn */
/* setup */
const data = {
    labels: days,
    datasets: [{
        data: retValue,
        fill: {
            target: 'origin',
            above: 'rgba(255, 99, 132, 0.7)',
            below: 'rgba(54, 162, 235, 0.7)'
        },
        borderColor: 'rgb(75, 192, 192)',
        hoverBackgroundColor: "rgba(0, 0, 0, 0.5)",
        tension: 0.3
    }]
};


/* config */
const config = {
    type: 'line',
    data: data,
    options: {
        scales: {
            yAxes: {
                suggestedMax: 3,
                suggestedMin: -3,
                ticks: {
                    beginAtZero: true,
                    stepSize: 1
                }
            }
        },

        plugins: {
            title: {
                display: true,
                text: ':: Daily Return Sep 2021 ::',
                font: {
                    size: 24,
                    weight: 'bold'
                }
            },
            legend: {
                display: false
            },
            tooltip: {
                enabled: true,
                displayColors: false,
                titleAlign: 'center',
                bodyAlign: 'right',
                callbacks: {
                    label: function(context) {
                        var value = retValue[context.dataIndex];

                        if (value > 0)
                            return '▲ ' + value.toFixed(2) + ' %';

                        else if (value < 0)
                            return '▼ ' + Math.abs(value).toFixed(2) + ' %';

                        else
                            return '0.00 %';
                    }
                }
            }
        }
    }
};


/* create chart */
var myChart = new Chart(
    document.getElementById('dailyReturn'),
    config
);
